### PR TITLE
#386: Factor out useful features of rank_object_enumerator.py and delete the rest

### LINKED ISSUE #386: Factor out useful features of rank_object_enumerator.py and delete the rest
`rank_object_enumerator.py` no longer works (nor is tested) in its current form, as reported by @pierrepebay as part of the review of PR #374 

However one of its routines is needed by `LBAF_app.py`; and this routine itself calls functions of `rank_object_enumerator.py`, and further. 

The goal of this issue is therefore to extract what is useful in `rank_object_enumerator.py` into a helper class (e.g. `lbsStatistics`) and then deprecate/remove the rest from the code base.

### PR DESCRIPTION
**Pull request by tlamonthezie**
Fixes #386 

This pull request includes :
- Move used functions from rank_object_enumerator to lbsStatistics
- Deprecate rank_object_enumerator using a log message
- Fix some problems with the BruteForceAlgorithm
- Remove duplicated code enumerating objects
- Refactor to use already populated Object instances instead of creating and reading dictionaries when calling  Statistics compute_*** methods for better understanding and code quality
- Make BruteForceAlgorithm work after some refactoring and code duplicates removals (use of lbsStatistics module to host the shared functions). You can tests if it works as expected using a configuration file that uses a BruteForceAlgorithm such as the following at `tests/config/conf_correct_brute_force.yml`

### CODE REVIEW

**Review by ppebay — CHANGES_REQUESTED** [review]

File: src/lbaf/IO/lbsStatistics.py [hunk 1]
```diff
@@ -153,14 +161,104 @@ def inverse_transform_sample(cmf):
...
+
+    # Return quantities of interest
+    return n_arrangements, works_min_max, arrangements_min_max
+
+
 def compute_function_statistics(population, fct) -> Statistics:
     """Compute descriptive statistics of a function over a population."""
 
     # Shorthand for NaN
     nan = math.nan
```
[the remark this answers is not in the document]
Comment (tlamonthezie) [reply src/lbaf/IO/lbsStatistics.py:258]: I remove it

File: src/lbaf/Model/lbsAffineCombinationWorkModel.py [hunk 2]
```diff
@@ -30,6 +30,18 @@ def __init__(self, parameters, lgr: Logger):
             self.__logger.info(
                 f"Upper bound for rank {k}: {v}")
 
+    def get_alpha(self):
+        """Get the alpha parameter"""
```
[the remark this answers is not in the document]
Comment (tlamonthezie) [reply src/lbaf/Model/lbsAffineCombinationWorkModel.py]: Yes the standard is to have a period as PEP257 said

**Review by ppebay — APPROVED** [review]
Thanks @tlamonthezie for changes

**Comment by tlamonthezie** [comment]
> Thanks @tlamonthezie for changes

Thank you @ppebay I let you tell me if I can merge (tomorrow).

**Pull request merged**